### PR TITLE
SHMEM_HEAP_TYPE constant is undefined, which causes compile error. Ch…

### PR DESCRIPTION
**Pull request by ireed**
…anged to read it as a system env var.

### CODE REVIEW

**Comment by ireed** [comment]
This commit is related to (and temporarily resolves) issue: https://github.com/openshmem-org/tests-mellanox/issues/16

File: verifier/basic/osh_basic_tc3.c [hunk 1]
```diff
@@ -49,7 +49,7 @@ static int memheap_type(void)
     if (memheap_type != MEMHEAP_ALLOC_UNKNOWN)
         return memheap_type;
 
-    p = getenv(SHMEM_HEAP_TYPE);
+    p = getenv("SHMEM_HEAP_TYPE");
```
Comment (miked-mellanox) [inline verifier/basic/osh_basic_tc3.c]: SHMEM_HEAP_TYPE constant is part of spec, cannot be just string
see here:
https://github.com/open-mpi/ompi/blob/master/oshmem/include/shmem.h.in#L76

Comment (ireed) [inline verifier/basic/osh_basic_tc3.c]: I see it in the implementation (code), but i dont see it defined (in the specs): https://github.com/openshmem-org/specification, 
[OpenSHMEM-1.3.pdf](https://github.com/openshmem-org/tests-mellanox/files/1238111/OpenSHMEM-1.3.pdf)

Is this constant unique to that flavor of openShmem?

Comment (jdinan) [inline verifier/basic/osh_basic_tc3.c]: This looks like a specification compliance bug in the OSHMEM implementation.  The ```SHMEM_HEAP_TYPE``` constant is not part of OpenSHMEM.  Added this to the open ticket filed with OMPI (https://github.com/open-mpi/ompi/issues/4098).

**Comment by jdinan** [comment]
@miked-mellanox Anything we can do to help with this PR?  Thanks!

File: verifier/basic/osh_basic_tc3.c (hunk 1 above)
Comment (jdinan) [reply verifier/basic/osh_basic_tc3.c]: I think this change is incorrect, although benign.  ```SHMEM_HEAP_TYPE``` is a string constant in the OMPI header file.  I think the right fix is to replace this with ```SHMEM_SYMMETRIC_HEAP_ALLOCATOR```, which is the value of the ```SHMEM_HEAP_TYPE``` constant.

Comment (miked-mellanox) [reply verifier/basic/osh_basic_tc3.c]: yep, seems you are right

in ompi shmem.h it is defined as ```#define SHMEM_HEAP_TYPE "SHMEM_SYMMETRIC_HEAP_ALLOCATOR"```

@ireed - can you please submit PR and use SPEC constant directly, thanks

Comment (jdinan) [reply verifier/basic/osh_basic_tc3.c]: Done -- #18.

**Pull request merged**